### PR TITLE
#50: Described release process in Developer Guide

### LINKED ISSUE #50: Developer Guide: Add how to release

### PR DESCRIPTION
**Pull request by ckunki**
Closes #50

### CODE REVIEW

File: doc/developer_guide/developer_guide.md [hunk 1]
```diff
@@ -66,3 +66,65 @@ Executing the integration tests requires the following environment variables to
...
 
+## Creating a Release
+
+### Release Preparation
+
+For preparing a release, the SAPIPY relies on Exasol's [python-toolbox](https://github.com/exasol/python-toolbox).
+
+The invocation depends on your setup:
+* When working in a poetry shell, you need to add one double-dash `--` argument to separate arguments to the nox-session `prepare-release`.
+* When calling `poetry` directly for one-time usage, then you need to add _two_ double-dashes `-- --` to terminate arguments to poetry and nox before arguments to the nox-session.
```
Comment (kaklakariada) [inline doc/developer_guide/developer_guide.md:77]: 😭

File: doc/developer_guide/developer_guide.md [hunk 2]
```diff
@@ -66,3 +66,65 @@ Executing the integration tests requires the following environment variables to
...
+* Create a branch, e.g. `prepare-release/1.2.3` starting from `main`
+* Update the version in files `pyproject.toml` and `version.py`
+* Update changes documentation
+  * Rename file `doc/unreleased.md` to `doc/changes_<version>.md` and add the current date as date of the release
+  * Create a new file `doc/unreleased.md`
+  * Update the file `doc/changelog.md`
+* Commit and push the changes
+* Create a pull request on GitHub
+
+Please note that the creating a pull request on GitHub requires
```
Comment (kaklakariada) [inline doc/developer_guide/developer_guide.md:93]:
```suggestion
Please note that creating a pull request on GitHub requires
```

File: doc/developer_guide/developer_guide.md [hunk 3]
```diff
@@ -66,3 +66,65 @@ Executing the integration tests requires the following environment variables to
...
+
+Nox session `prepare-release` will
+* Create a branch, e.g. `prepare-release/1.2.3` starting from `main`
+* Update the version in files `pyproject.toml` and `version.py`
+* Update changes documentation
+  * Rename file `doc/unreleased.md` to `doc/changes_<version>.md` and add the current date as date of the release
+  * Create a new file `doc/unreleased.md`
+  * Update the file `doc/changelog.md`
+* Commit and push the changes
+* Create a pull request on GitHub
```
Comment (kaklakariada) [inline doc/developer_guide/developer_guide.md:91]: This duplicates the documentation in the python-toolbox. Maybe just add a link.

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Review by kaklakariada — APPROVED** [review]

File: doc/developer_guide/developer_guide.md [hunk 4]
```diff
@@ -66,3 +66,65 @@ Executing the integration tests requires the following environment variables to
...
+The invocation depends on your setup:
+* When working in a poetry shell, you need to add one double-dash `--` argument to separate arguments to the nox-session `prepare-release`.
+* When calling `poetry` directly for one-time usage, then you need to add _two_ double-dashes `-- --` to terminate arguments to poetry and nox before arguments to the nox-session.
+
+```shell
+poetry run nox -s prepare-release -- -- <version>
+```
+
+Nox session `prepare-release` will
+* Create a branch, e.g. `prepare-release/1.2.3` starting from `main`
```
Comment (ahsimb) [inline doc/developer_guide/developer_guide.md:84]: I think, in this case, we should be in the main branch locally.

**Review by kaklakariada — APPROVED** [review]

**Review by ahsimb — APPROVED** [review]

**Pull request merged**